app/services/proforma_service/convert_exercise_to_task.rb

```diff
@@ -22,12 +22,24 @@ def create_task
         {
           title: @exercise.title,
           description: @exercise.description,
-          internal_description: @exercise.instructions,
+          internal_description: nil,
+          # proglang: proglang, where can we get this information?
```

Comment:
What about the `proglang` attribute (here and in the specs)? For the programming language, you can probably use the `execution_environment.file_type.editor_mode`. That's the best information we have.

Otherwise the `execution_environment.docker_image` might be parsed using the following regex `^(?:openhpi\/co_execenv_(?<language>[^:]*):(?<version>[^-]*)(?>-.*)?)$`, which will yield two named results `language` and `version`.

Reply:
> Otherwise the `execution_environment.docker_image` might be parsed using the following regex `^(?:openhpi\/co_execenv_(?<language>[^:]*):(?<version>[^-]*)(?>-.*)?)$`, which will yield two named results `language` and `version`.

this would be really nice - does it work with production values? The dev-seeds have a lot like this: `hklement/ubuntu-coffee:latest`

Reply:
Yes, it should. We mainly use the following images:


openhpi/co_execenv_java:8-antlr
openhpi/co_execenv_java:17
openhpi/co_execenv_ruby:2.5
openhpi/co_execenv_node:0.12
openhpi/co_execenv_python:3.4
openhpi/co_execenv_python:3.4-rpi-web
openhpi/co_execenv_python:3.7-ml
openhpi/co_execenv_python:3.8
openhpi/co_execenv_r:4

Unfortunately, the seed data are not up-to-date with regard to the Docker images used.